Lesson5.ipynb

```diff
@@ -29,6 +32,384 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Normal loop implementation\n",
+    "evenSquare = []\n",
+    "for x in range(10):\n",
+    "    if x % 2 == 0:\n",
+    "        evenSquare.append(x**2)\n",
+    "        \n",
+    "print(evenSquare)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# List comprehensions implementation\n",
+    "evenSquare = [x**2 for x in range(10) if x % 2 == 0]\n",
+    "print(evenSquare)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Example 2: Create a tuple that have all different numbers combination between 2 array range from 0 to 4."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Normal loop implementation\n",
+    "combination = []\n",
+    "for x in range(5):\n",
+    "    for y in range(5):\n",
+    "        if (x != y):\n",
+    "            combination.append((x, y))\n",
+    "            \n",
+    "print(combination)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# List comprehensions implementation\n",
+    "combination = [(x, y) for x in range(5) for y in range(5) if x != y]\n",
+    "print(combination)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Example 3: Given 2-dimentional list (like a table), flatten it (to a normal list)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "table = [[1, 2, 3],\n",
+    "         [4, 5, 6],\n",
+    "         [7, 8, 9]]\n",
+    "flatten = [num for row in table for num in row]\n",
+    "print(flatten)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Example 4: Given 2-dimentional list (like a table), transpose rows and columns using Nested List Comprehensions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "transpose = [[row[num] for row in table] for num in range(len(table[0]))]\n",
+    "for display in transpose:\n",
+    "    print(display)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Sorting\n",
+    "\n",
+    "### Ascending - from small to big, from low to high, from a to z\n",
+    "\n",
+    "By default, all sorting are in ascending order.\n",
+    "\n",
+    "**sorted(list)** will return a new sorted list, leaving the original list unaffected.\n",
+    "\n",
+    "**list.sort()** will sorts the list **in-place**. In case list is muted, it will return None.\n",
+    "\n",
+    "### Descending - in reverse of Ascending\n",
+    "\n",
+    "Both sort() and sorted() accept additional parameter as reverse in boolean type.\n",
```

Comment:
"Both sort() and sorted() accept additional parameter as reverse in boolean type." should read "Both sort() and sorted() accept an additional parameter for reverse in boolean type."